tmt/steps/prepare/brand_new_allmighty_install/providers/__init__.py

```diff
@@ -29,15 +32,28 @@ class DownloadError(GeneralError):
 class ArtifactInfo:  # TODO: Gather artifact metadata
     name: str
     type: ArtifactType
+    arch: str
 
     def __str__(self) -> str:
         return self.name
 
 
 class ArtifactProvider(ABC):
-    def __init__(self, logger: tmt.log.Logger, artifact_id: str):
+    def __init__(self, logger: tmt.log.Logger, artifact_id: str, api_url: str):
         self.logger = logger
         self.artifact_id = self._parse_artifact_id(artifact_id)
+        self.api_url = api_url
+        self._session: ClientSession = self._initialize_session()
+
+    @abstractmethod
+    def _initialize_session(self) -> ClientSession:
```

Comment:
I would not generalize this and `_call_api`, instead would move it to the koji provider